pandas/_libs/tslibs/parsing.pyx

```diff
@@ -44,13 +50,73 @@ class DateParseError(ValueError):
 
 _DEFAULT_DATETIME = datetime(1, 1, 1).replace(hour=0, minute=0,
                                               second=0, microsecond=0)
+_DEFAULT_TZINFO = _DEFAULT_DATETIME.tzinfo
 
 cdef:
     object _TIMEPAT = re.compile(r'^([01]?[0-9]|2[0-3]):([0-5][0-9])')
 
     set _not_datelike_strings = {'a', 'A', 'm', 'M', 'p', 'P', 't', 'T'}
 
 # ----------------------------------------------------------------------
+DEF delimiters = b' /-\\'
+DEF MAX_DAYS_IN_MONTH = 31
+DEF MAX_MONTH = 12
```

Comment:
any particular reason these are DEF instead of cdef?

Reply:
There's no possibility to declare `cdef const int FOO = 5` in Cython (thus C compilers cannot do some optimizations). But that's a minor reason :)

The main reason is it's currently impossible to (in a readable manner) declare this C equivalent in Cython:
```C
static const char delimiters[] = " /-\\";
```
so that C array size would be determined by compiler (and would be automatically NULL-terminated).
So we decided to go using `DEF` here.

Reply:
> so that C array size would be determined by compiler (and would be automatically NULL-terminated).

I take it that is costly?

Mixing and matching syntax makes these files less accessible for readers/maintainers (e.g. I just recently learned about `DEF` and I work in these files all the time).  This isn't a deal-breaker, but to the extent you can keep things accessible to a cython-newbie, please try to do so.

Reply:
It is not performance cost (that difference is probably negligible), just compare it to alternative:
```cython
cdef char delimiters[5]
delimiters[:] = [b' ', b'/', b'-', b'\\', b'\0']
```
Does this look easier to you?.. And imagine that you'd need to add another delimiter - how big is probability of adding it _after_ `NULL` terminator thus having no effect?

Reply:
you cannot do this in one line? I agree with @jbrockmendel here we want to make the simpest possible cython, keeping the number of idioms small; we have a very complex code base and this is just expanding it (now if we want to do something everywhere well that is a different story, but that is out of scope here)

Reply:
> And imagine that you'd need to add another delimiter - how big is probability of adding it after NULL terminator thus having no effect?

This sounds like you're saying maintenance will be full of pitfalls regardless of the implementation.  This makes me more inclined to say "just pass a `format` kwarg"

> `static const char delimiters[] = " /-\\";`

1) could you get this using `cdef extern from *:` idiom?
2) if this usage is really important, it might be worth opening an issue on the cython tracker to get it implemented.

Reply:
`cdef extern from *` doesn't define stuff, it only declare it and assume that its defintion would happen somewhere in included files, so I cannot use it.

For now I ended up with:
```cython
cdef:
    const char* delimiters = " /-\\."
    int MAX_DAYS_IN_MONTH = 31, MAX_MONTH = 12
```